libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -241,7 +241,7 @@ export const [hasActiveLink, setActiveLink, getActiveLink] = (() => {
     () => activeLinkFound,
     (val) => { activeLinkFound = !!val; },
     (area) => {
-      if (hasActiveLink() || !(area instanceof HTMLElement)) return null;
+      if (hasActiveLink() || !(area instanceof HTMLElement) || area.querySelector('a').dataset.modalHash) return null;
```

Comment:
An area can contain multiple links. If the first link happens to open a modal, the others would be ignored. The logic should probably be added lower, to check that when a candidate for an active link is found, it doesn't open a modal.